spock-core/src/main/java/org/spockframework/runtime/ValueRecorder.java

```diff
@@ -27,6 +27,7 @@
  */
 public class ValueRecorder {
   private final ArrayList<Object> values = new ArrayList<Object>();
+  private final Stack<Integer> startedRecordings = new Stack<Integer>();
```

Comment:
http://docs.oracle.com/javase/8/docs/api/java/util/Stack.html
> A more complete and consistent set of LIFO stack operations is provided by the Deque interface and its implementations, which should be used in preference to this class.

And Stack is based on Vector:
> Unlike the new collection implementations, Vector is synchronized. If a thread-safe implementation is not needed, it is recommended to use ArrayList in place of Vector.